### PR TITLE
Closes #1880 Avoid warnings for latest R version

### LINKED ISSUE #1880: Bug: warnings in tests related to deprecated tidyselect functionality
### What happened?

**Very important note: please attempt to reproduce this in your own RStudio platform where you do your admiral development! The problem may be due to something I have done incorrectly!**

**NOTE: instead of using renv::restore(), I used renv::restore(lockfile='./renv/profiles/4.2/renv.lock') because the renv::restore() seems to load the R4.0.5 packages; I am opening another Issue about that (see Issue pharmaverse/admiralci#93).**

When using R 4.2.1, some of the tests throw warnings about tidyselect. I have attached the test results.
They seem to fall into two groups, with multiple calls for each group:

```
Warning (test-derive_var_last_dose_amt.R:38:3): derive_var_last_dose_amt Test 1: works as expected
Using an external vector in selections was deprecated in tidyselect 1.1.0.
i Please use `all_of()` or `any_of()` instead.
```
```
Warning (test-derive_vars_query.R:33:3): derive_vars_query Test 1: Derive CQ and SMQ variables with two term levels
Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.
i Please use `"VAR_PREFIX_NAM"` instead of `.data$VAR_PREFIX_NAM`
```

[test.txt](https://github.com/pharmaverse/admiral/files/11264941/test.txt)

### Session Information

image

### Reproducible Example

- Create a new project in RStudio (R v4.2.1) with admiral's GitHub link. 
- Switch to the devel branch.
- Do renv::restore()  NOTE: ensure the packages to be loaded follow the versions specified for R4.2! If not, do renv::restore(lockfile='./renv/profiles/4.2/renv.lock') instead.
- Do devtools::test()

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_param_tte.R [hunk 1]
```diff
@@ -752,7 +752,8 @@ extend_source_datasets <- function(source_datasets,
         full_join(
           mutate(by_groups, temp_dummy = 1),
           mutate(source_datasets[[i]], temp_dummy = 1),
-          by = "temp_dummy"
+          by = "temp_dummy",
+          relationship = "many-to-many"
```
Comment (zdz2101) [inline R/derive_param_tte.R:756]: Do we have a "history" or issue this particular modification fixes? I'm sure it was done with good reason but based on #1880 only the other changes seemed necessary

Comment (bundfussr) [reply R/derive_param_tte.R:756]: This is necessary to fix
```
Detected an unexpected many-to-many relationship between `x` and `y`.
i Row 1 of `x` matches multiple rows in `y`.
i Row 1 of `y` matches multiple rows in `x`.
i If a many-to-many relationship is expected, set `relationship = "many-to-many"` to silence this warning.
Backtrace:
  1. testthat::expect_error(...)
       at test-derive_param_tte.R:638:2
````

**Review by zdz2101 — APPROVED** [review]
